modmesh/pilot/airfoil/__init__.py

```diff
@@ -0,0 +1,264 @@
+
+          y_t = 5t(0.2969 \sqrt{x} - 0.1260 x - 0.3516 x^2 + 0.2843 x^3
+                 - 0.1015 x^4)
+
+        Zero (close) trailing edge:
+
+        .. math::
+
+          y_t = 5t(0.2969 \sqrt{x} - 0.1260 x - 0.3516 x^2 + 0.2843 x^3
+                 - 0.1036x^4)
+
+        :param xc: Location along the x direction
+        :return: The y value (positive)
+        """
+        xh = np.sqrt(xc)
+        x2 = xc * xc
+        x3 = x2 * xc
+        x4 = x3 * xc
+        a0 = self.A0
+        a1 = self.A1
+        a2 = self.A2
+        a3 = self.A3
+        a4 = self.A4O if self.open_trailing_edge else self.A4C
+        t = self.thick
+        yt = 5.0 * t * (a0 * xh + a1 * xc + a2 * x2 + a3 * x3 + a4 * x4)
+        return yt
+
+    def calc_ul(self, xc):
+        r"""
+        Calculate the coordinates (xu, yu) and (xl, yl) (u for upper and l for
+        lower) at the corresponding location on the center/camber.
+
+        For asymmetric camber (pos and camber are non-zero), the mean camber
+        line is calculated (:math:`m` is the maximum camber, :math:`p` is the
+        position of the maximum camber):
+
+        .. math:
+
+          y_c = \frac{m}{p^2}(2px - x^2), \quad 0 \le x \le p \\
+          y_c = \frac{m}{(1-p)^2}(1 - 2p + 2px - x^2), \quad p \le x \le 1
+
+        The thickness needs to be perpendicular to the camber line, the upper
+        and lower profile should be calculated by:
+
+        .. math:
+
+          x_u = x - y_t\sin\theta , \; y_u = y_c + y_t\cos\theta \\
+          x_l = x + y_t\sin\theta , \; y_l = y_c - y_t\cos\theta
+
+        The angle :math:`\theta` should be determined by:
+
+        .. math:
+
+          \theta = \arctan\frac{\mathrm{d}y_c}{\mathrm{d}x} \\
+          \frac{\mathrm{d}y_c}{\mathrm{d}x} = \frac{2m}{p^2}(p-x),
+            \quad 0 \le x \le p \\
+          \frac{\mathrm{d}y_c}{\mathrm{d}x} = \frac{2m}{(1-p)^2}(p-x),
+            \quad p \le x \le 1
+
+        :param xc: x coordinate at center/camber
+        :return: xu, yu, xl, yl
+        """
+        yt = self.calc_yt(xc)
+        pos = self.pos
+        if pos == 0:
+            return xc, yt, xc, -yt
+        else:
+            camber = self.camber
+            if xc <= pos:
+                xc1 = xc
+                yc1 = camber / np.power(pos, 2) * xc1 * (2 * pos - xc1)
+                dyc1_dx = camber / np.power(pos, 2) * (2 * pos - 2 * xc1)
+                yc = yc1
+                dyc_dx = dyc1_dx
+            else:
+                xc2 = xc
+                yc2 = camber / np.power(1 - pos, 2) * (1 - 2 * pos + xc2) * (
+                        1 - xc2)
+                dyc2_dx = camber / np.power(1 - pos, 2) * (2 * pos - 2 * xc2)
+                yc = yc2
+                dyc_dx = dyc2_dx
+
+            theta = np.arctan(dyc_dx)
+
+            return (xc - yt * np.sin(theta), yc + yt * np.cos(theta),
+                    xc + yt * np.sin(theta), yc - yt * np.cos(theta))
+
+    def calc_points(self, npoint):
```

Comment:
`Naca4.calc_points()` samples the point on the profile and return an array for the sampled coordinates.